lib/dao.js

```diff
@@ -1597,6 +1598,22 @@ DataAccessObject._coerce = function(where) {
 
       continue;
     }
+
+    if (p.match(/\./)) {
+      var model = p.split('.')[0];
+      var prop = p.split('.').slice(1);
+
+      if (props[model]) {
+        var clause = {};
+        clause[prop] = where[p];
+        where[p] = Array.isArray(props[model].type) ?
+          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
+          self._coerce(clause, props[model].type.definition.properties)[prop];
```

Comment:
I am not sure if I follow this correctly. Was there a typo on `type.difinition.properties`? Because the actual code points to **type.definition.properties**.